mod-source-record-storage-server/src/main/java/org/folio/services/util/AdditionalFieldsUtil.java

```diff
@@ -152,34 +151,79 @@ public static boolean addControlledFieldToMarcRecord(Record record, String field
   /**
    * remove field from marc record
    *
-   * @param record record that needs to be updated
-   * @param field  tag of the field
+   * @param record   record that needs to be updated
+   * @param field    tag of the field
+   * @param subfield subfield of the field
+   * @param value    value of the field
    * @return true if succeeded, false otherwise
    */
-  public static boolean removeField(Record record, String field) {
-    boolean result = false;
+  public static boolean removeField(Record record, String field, char subfield, String value) {
+    boolean isFieldRemoveSucceed = false;
     try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
       if (record != null && record.getParsedRecord() != null && record.getParsedRecord().getContent() != null) {
         MarcReader reader = buildMarcReader(record);
         MarcWriter marcStreamWriter = new MarcStreamWriter(new ByteArrayOutputStream());
         MarcJsonWriter marcJsonWriter = new MarcJsonWriter(baos);
         if (reader.hasNext()) {
           org.marc4j.marc.Record marcRecord = reader.next();
-          VariableField variableField = marcRecord.getVariableField(field);
-          if (variableField != null) {
-            marcRecord.removeVariableField(variableField);
+          List<VariableField> variableFields = marcRecord.getVariableFields(field);
+          for (VariableField variableField : variableFields) {
+            if (StringUtils.isEmpty(value)) {
+              marcRecord.removeVariableField(variableField);
+              isFieldRemoveSucceed = true;
+              break;
+            } else {
```

Comment:
@Aliaksandr-Fedasiuk  I think we can use here "else-if" instead if inside else.

Reply:
Thank you, @VRohach , completed